src/client/mapblock_mesh.cpp

```diff
@@ -969,9 +972,13 @@ video::SColor encode_light(u16 light, u8 emissive_light)
 		r = day * 255 / sum;
 	else
 		r = 0;
+	if (r < 0xF0 && r&0b1100)
```

Comment:
What's the reasoning behind this?

Reply:
This is literally rounding code.

Reply:
We've got numbers that should be 0xFF at most (that is the requirement of SColor constructor that is (and was) used). We're cutting off the lower 4 bits later. So, if the 4 significant bits are 0xF, they can't represent a higher number anyway. If they're not, it can be rounded upwards if needed. Then two significant bits of the dropped tail are checked, and if one of them is set, ceil it.

Reply:
Lambda is overkill for two usages one right after the other tbh.